devito/dle/backends/basic.py

```diff
@@ -89,11 +89,10 @@ def _create_elemental_functions(self, state, **kwargs):
 
                 # Retrieve tensor arguments
                 for i in fsymbols:
-                    if i.is_SymbolicFunction:
-                        handle = "(%s*) %s" % (c.dtype_to_ctype(i.dtype), i.name)
-                    else:
-                        handle = "%s_vec" % i.name
-                    args.append((handle, i))
+                    if i.is_TensorFunction:
+                        args.append(("(%s*)%s" % (c.dtype_to_ctype(i.dtype), i.name), i))
+                    elif i.is_SymbolicData:
+                        args.append(("%s_vec" % i.name, i))
```

Comment:
turns out this is the string for the function call, not the function declaration
we could definitely add it along with ``decl`` ``ccode`` etc, but not part of this PR